srv/modules/runners/disk_led.py

```diff
@@ -0,0 +1,181 @@
+"""
+from __future__ import absolute_import
+import logging
+import re
+import salt.client
+
+
+log = logging.getLogger(__name__)
+
+
+def _process(hostname, device_name, led, status):
+    """
+    Turn on/off the specified LED of the disk on the given host.
+    """
+    local_client = salt.client.LocalClient()
+    device_file = '/dev/{}'.format(device_name)
+    status = 'on' if status in [True, 'on'] else 'off'
+
+    # Get the configuration of the command to turn on/off the ident
+    # or fault LED on the target host.
+    cmd_cfg = local_client.cmd(
+        hostname, 'pillar.get', ['disk_led:cmd'],
+        tgt_type='compound')[hostname]
+
+    print("Turning {} the {} LED of disk '{}' on host '{}' ...".format(
+        status, led, device_file, hostname))
+    _cmd_run(hostname, cmd_cfg[led][status].format(device_file=device_file))
+
+
+def _cmd_run(hostname, cmd):
+    """
+    Run the specified command line.
+    .. note:: This function has been introduced for easier unit testing.
+    """
+    local_client = salt.client.LocalClient()
+    local_client.cmd(hostname, 'cmd.run', [cmd], tgt_type='compound')
+
+
+def osd(osd_id, device_type, led, status):
+    """
+    Manage the storage enclosure LED by OSD identifier.
+    :param osd_id: The OSD id, e.g.
+        - 4
+        - osd.1
+    :type osd_id: str
+
+    :param device_type: The device to use. This can be:
+        - primary
+        - journal
+        - db
+        - all
+    :type device_type: str
+    :param led: The LED to be processed. This can be 'ident' or 'fault'.
+    :type led: str
+    :param status: Set to 'on' to switch on the light, otherwise 'off'.
+    :type status: str
+    """
+    osd_id = int(re.sub(r'^osd\.', '', str(osd_id)))
+    local_client = salt.client.LocalClient()
+
+    # Get OSD information from all storage nodes.
+    osd_infos = local_client.cmd(
+        'I@roles:storage', 'osd.devices_details', [], tgt_type='compound')
+    # Process OSD information.
+    # {
+    #   "admin.ceph": {},
+    #   "mon1.ceph": {},
+    #   "data1.ceph": [{
+    #     "cluster": "ceph",
+    #     "id": 5,
+    #     "name": "osd.5",
+    #     "hostname": "data1",
+    #     "type": "bluestore",
+    #     "primary": {
+    #       "rotational": true,
+    #       "device_name": "vdg",
+    #       "device_file": "/dev/vdg"
+    #     },
+    #     "journal": {
+    #       "rotational": null,
+    #       "device_name": null,
+    #       "device_file": null
+    #     },
+    #     "db": {
+    #       "rotational": null,
+    #       "device_name": null,
+    #       "device_file": null
+    #     }
+    #   }]
+    # }
+    for hostname in osd_infos:
+        filtered_osd_infos = list([x for x in osd_infos[hostname]
+                                   if x['id'] == osd_id])
+        try:
+            osd_info = filtered_osd_infos[0]
+        except IndexError:
+            continue
+        device_types = ['primary', 'journal', 'db'] if device_type == 'all' \
```

Comment:
Imho if a oneline `if` doesn't fit on one line, use the traditional syntax? This could also simpy merge into the processing below:
``` python
if device_type == 'all':
    for t in ['primary', 'journal', 'db']:
        get_device_name_and_process(t)
else:
    get_device_name_and_process(device_type)